File: NapakalakiRuby/lib/dice.rb
# By: Miguel Robles Urquiza
#     Jesús Sánchez de Lechina Tejada
 require "singleton"

module NapakalakiGame

  class Dice
    
    def inicialize
      
    end
    
    include Singleton

    def nextNumber
        return 1 + rand(6)
    end
    
  end
end

File: Napakalaki/src/NapakalakiGame/Dice.java
package NapakalakiGame;
import java.util.Random;


/**
 *
 * @author Jesús Sánchez de Lechina Tejada & Miguel Ángel Robles Urquiza
 */
public class Dice {
    
  private static  Dice instance = null;
 
  private Dice() { }
 
  public static Dice getInstance() {
 
        if (instance == null)
            instance = new Dice();
        return instance;
    }
  public int nextNumber() {
        Random rnd = new Random();    
        // rnd.nextInt devuelve un int aleatorio, el 6 dice la longitud del rango y 
        //el 0 es el valor inicial
        return (int)(rnd.nextInt() * 6 + 0);
    }
    
}
